File: dcp-solutions/2024-11-25/solution.py
```python
def min_columns_to_remove(matrix):
    rows = len(matrix)
    cols = len(matrix[0])
    
    # Sort each row's columns lexicographically
    for row in matrix:
        row.sort()
        
    max_count = 0
    prev_char = None
    
    for col in range(cols - 1, -1, -1):
        if prev_char is not None and ord(prev_char) > ord(matrix[0][col]):
            break
        else:
            prev_char = matrix[0][col]
            max_count += 1
            
    return cols - max_count
```

File: dcp-solutions/2024-11-25/solution.go
```go
package main

import (
	"sort"
)

func minColumnsToRemove(matrix [][]byte) int {
	var rows, cols = len(matrix), len(matrix[0])
	sort.SliceStable(matrix, func(i, j int) bool {
		for k := range matrix[i] {
			if matrix[i][k] != matrix[j][k] {
				return matrix[i][k] < matrix[j][k]
			}
		}
		return false
	})

	for col := range matrix[0] {
		for row := 1; row < rows; row++ {
			if matrix[row-1][col] > matrix[row][col] {
				return cols - 1
			}
		}
	}
	return 0
}

func main() {}
```